test/functional/plugin/health_spec.lua

```diff
@@ -77,6 +77,14 @@ describe(':checkhealth', function()
       return vim.api.nvim_win_get_config(0).relative
     ]])
     )
+
+    -- Prevent checkhealth floating window closing after using gO. See #34784
+    command('checkhealth full_render')
+    local win = api.nvim_get_current_win()
+    api.nvim_win_set_cursor(win, { 5, 1 })
+    n.feed('gO')
```

Comment:
feed() is async. that means the assertions below this could have false positives? maybe `command('norm gO')` instead ?

Reply:
But `nvim_win_is_valid` is not async.